### DISCUSSION TITLE
JavaScript SDK help

### DISCUSSION THREAD

**Discussion by GergoEB** [pocketbase/pocketbase#728] [Q&A] [answered]
Im trying to make a project with vuejs and pocketbase handling users, and im fairly new the javascript (good in java so i know some), some questions i cannot find answers to are:
1 - How does pocketbase app secure itself? what is stopping it from being logged into from anywhere
2 - How do I get a user id from the cookie? i load cookie from client then `client.authStore.loadFromCookie()`, where to from there?
3 - In vuejs can i place the `const client = new PocketBase('http://127.0.0.1:8090');` in main.js or do i  use it only where i need it

the only tutorial that somewhat answeres these questions uses typescript and I dont know how much of it is relevant to javascript

**Comment by ganigeorgiev** [comment, marked as answer]
> 1 - How does pocketbase app secure itself? what is stopping it from being logged into from anywhere

We use JWT tokens for authentication. Each collection has API rules to control the access for the list, view, create, update, delete actions. You could read more about this in https://pocketbase.io/docs/manage-collections/

---

> 2 - How do I get a user id from the cookie? i load cookie from client then client.authStore.loadFromCookie(), where to from there?

 You need cookies only if you are dealing with server-side rendering (SSR). If you are working on a client-side only application (eg. SPA), the default LocalStorage `AuthStore` is recommended. You can read more about the `AuthStore` and how to access the model and token after authentication in https://github.com/pocketbase/js-sdk#authstore.

---

> 3 - In vuejs can i place the const client = new PocketBase('http://127.0.0.1:8090'); in main.js or do i use it only where i need it

This is hard to generalize because it depends on your project structure. You can use a single client instance for the entire lifecycle of the application (this is suitable for SPA) or if you are using server-side rendering - you can instantiate a new client for each request (you can find some ssr examples in https://github.com/pocketbase/js-sdk#ssr-integration).

**Reply by GergoEB** [reply]
That was helpful, how do I get a user id from the cookie token? Is this even possible

**Reply by ganigeorgiev** [reply]
After you've loaded the cookie with `client.authStore.loadFromCookie()` you can access the auth model id by doing something like `client.authStore.model.id`.
